lib/android/src/main/java/com/airbnb/android/react/maps/AirMapView.java

```diff
@@ -846,4 +853,64 @@ public void onPanDrag(MotionEvent ev) {
         WritableMap event = makeClickEventData(coords);
         manager.pushEvent(context, this, "onPanDrag", event);
     }
+    @Override
+    public void onIndoorBuildingFocused() {
+      IndoorBuilding building = this.map.getFocusedBuilding();
+      if (building != null) {
+        List<IndoorLevel> levels = building.getLevels();
+        int index = 0;
+        WritableArray levelsArray = Arguments.createArray();
+        for (IndoorLevel level : levels) {
+          WritableMap levelMap = Arguments.createMap();
+          levelMap.putInt("index", index);
+          levelMap.putString("name", level.getName());
+          levelMap.putString("shortName", level.getShortName());
+          levelsArray.pushMap(levelMap);
+    		}
+
+        WritableMap event = Arguments.createMap();
+        WritableMap indoorBuilding = Arguments.createMap();
+
+        indoorBuilding.putArray("levels", levelsArray);
+        indoorBuilding.putInt("defaultLevelIndex", building.getActiveLevelIndex());
+        indoorBuilding.putBoolean("underground", building.isUnderground());
+
+        event.putMap("indoorBuilding", indoorBuilding);
+
+        manager.pushEvent(context, this, "onIndoorBuildingFocused", event);
+
+      }
+    }
+
+    @Override
+    public void onIndoorLevelActivated(IndoorBuilding building) {
+      int activeLevelIndex = building.getActiveLevelIndex();
+
+      IndoorLevel level = building.getLevels().get(activeLevelIndex);
+
+      WritableMap event = Arguments.createMap();
+      WritableMap indoorlevel = Arguments.createMap();
+
+      indoorlevel.putInt("activeLevelIndex", activeLevelIndex);
+      indoorlevel.putString("name", level.getName());
+      indoorlevel.putString("shortName", level.getShortName());
+
+      event.putMap("indoorlevel", indoorlevel);
+
+      manager.pushEvent(context, this, "onIndoorLevelActivated", event);
+    }
+
+    public void setIndoorActiveLevelIndex(int activeLevelIndex) {
+      IndoorBuilding building = this.map.getFocusedBuilding();
+      if (building != null) {
+        try {
+          IndoorLevel level = building.getLevels().get(activeLevelIndex);
+          if (level != null) {
+            level.activate();
+          }
+        } catch(IndexOutOfBoundsException ignored) {
```

Comment:
instead of swallowing this exception, check the array bounds first before accessing the element(s)